### ISSUE TITLE
Calendar month has padding that keeps it from aligning with content around it

### ISSUE THREAD

**Issue by mcoker** [patternfly/patternfly#5272]
looking at the [calendar month examples](https://www.patternfly.org/v4/components/calendar-month), the calendar component has padding around it that will provide space around it wherever it is used. Typically our components don't have padding like that and extend to the edges of wherever they're placed[1], if that were the case here, the thing the calendar is placed inside of would provide that padding instead of the calendar creating the padding itself.

You can see here how the calendar content does not align with the content above it because of the calendar's built-in padding.

<img width="586" alt="Screenshot 2022-12-01 at 4 55 47 PM" src="https://user-images.githubusercontent.com/35148959/205176846-e6cb9ca8-c84f-4da0-8465-55e26209bac9.png">

<img width="530" alt="Screenshot 2022-12-01 at 4 59 26 PM" src="https://user-images.githubusercontent.com/35148959/205176973-2fe5680c-07a8-4a36-b10e-f71e48b239ab.png">

#### Here is what it looks like if the calendar's default padding is removed

<img width="527" alt="Screenshot 2022-12-01 at 4 59 53 PM" src="https://user-images.githubusercontent.com/35148959/205177286-9c73a555-23be-4e65-b246-13d1c3eb54a7.png">

#### Here is what the calendar looks like in the date picker popover
Note that the popover's built-in padding has been removed (using a no-padding variation), and the calendar component's space is creating the padding.

<img width="751" alt="Screenshot 2022-12-01 at 4 52 54 PM" src="https://user-images.githubusercontent.com/35148959/205176217-30a4dbb9-c164-4bd2-8868-f4e4e18bc4f7.png">

#### Here is what it looks like if the calendar component's padding is removed, and the popover creates the padding

<img width="639" alt="Screenshot 2022-12-01 at 4 53 19 PM" src="https://user-images.githubusercontent.com/35148959/205176156-5d1fe0d9-9d2b-484f-8edf-21e5274b53b6.png">

One gotcha with the padding is that the calendar month's built-in padding has a larger top padding than bottom padding, to help balance the top/bottom whitespace visually since there is extra padding below each date. That extra space is so we can draw a circle around the date when it's selected. So if none of the dates in the bottom row are selected, it looks like the spacing is imbalanced. Here's what it looks like with a date selected to illustrate what's going on there.

<img width="667" alt="Screenshot 2022-12-05 at 11 23 54 AM" src="https://user-images.githubusercontent.com/35148959/205702775-20e0022b-c274-4abf-bd9a-aa514b6958a8.png">

I think that would be more flexible and more inline with our principles around padding/spacing since the calendar month can go anywhere. On its own on the page, it looks a bit odd with the padding, and looks better without the padding IMO. If it needed to go in a box - like a popover or card or panel (dropdown), all of those components provide the inner padding for their contents.

cc @mcarrano @mceledonia @srambach @mattnolting - wdyt?

[1]. Unless the component has some kind of visible boundary around its outermost container - a border or shadow (like a card) or maybe a unique background color that creates the box shape of the container.

**Comment by mattnolting** [comment]
@mcoker I completely agree. I don't see an issue with keeping the padding variables and setting them to `0`, just in case there is an edge case where the calendar needs to be adjusted. Also could use the padding utility for that, so I'm not opposed to removing it altogether.

**Comment by srambach** [comment]
Absolutely agree - afaik the general rule has been pad the component, margin the children, and a component takes up available space.
